test/ibmq/test_ibmq_job.py

```diff
@@ -378,7 +380,7 @@ def test_retrieve_failed_job_device(self, provider):
         qobj.experiments[1].instructions[1].name = 'bad_instruction'
 
         job = backend.run(qobj)
-        with self.assertRaises(JobError):
+        with self.assertRaises(IBMQJobApiError):
```

Comment:
This should be an `IBMJobFailureError`. Retrieving a failed job shouldn't cause api error